### PR TITLE
make training data accessible for callbacks (e.g., on_epoch_end)

### PR DESCRIPTION
**Pull request by twoertwein**
make it possible to access training data through callbacks. Can be used to calculate the actual loss of the training set for each epoch

### CODE REVIEW

**Comment by carlthome** [comment]
How does this work with `fit_generator`? Just curious!
